Src/AutoNSubstitute/SubstituteBuilder.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute
+{
+    /// <summary>
+    /// Creates a substitute in response to the <see cref="SubstituteRequest"/>.
+    /// </summary>
+    /// <remarks>
+    /// This class serves as a fixture customization, responding to explicit substitute requests in the form of the
+    /// <see cref="SubstituteRequest"/> instances. The actual construction of the substitute is delegated to a 
+    /// substitute constructor, responsible for invoking an appropriate constructor for the target type with
+    /// automatically generated arguments.
+    /// </remarks>
+    public class SubstituteBuilder : ISpecimenBuilder
+    {
+        private readonly ISpecimenBuilder substituteConstructor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SubstituteBuilder"/> class with the specified 
+        /// <paramref name="substituteConstructor"/>.
+        /// </summary>
+        /// <param name="substituteConstructor">
+        /// An <see cref="ISpecimenBuilder"/> responsible for creating a substitute instance from a requested 
+        /// <see cref="Type"/>.
+        /// </param>
+        public SubstituteBuilder(ISpecimenBuilder substituteConstructor)
+        {
+            if (substituteConstructor == null)
+            {
+                throw new ArgumentNullException("substituteConstructor");
+            }
+
+            this.substituteConstructor = substituteConstructor;
+        }
+
+        /// <summary>
+        /// Returns an <see cref="ISpecimenBuilder"/> responsible for creating a substitute instance based on a target
+        /// <see cref="Type"/> type.
+        /// </summary>
+        public ISpecimenBuilder SubstituteConstructor 

```

Comment:
> `SubstituteBuilder` property in a `SubstituteBuilder` class would be confusing.

Good point. I completely agree. I hadn't really had my eye on the slightly bigger picture, there :flushed: 

Although I see your point about giving the class itself a more descriptive name, I don't mind `SubstituteBuilder` as a class name, because this is, after all, the 'top' class responsible for building substitutes. In that case, I prefer `SubstituteFactory` as the property name.